src/a-runtime-console/kubernetes/ui/pipeline/input-action-dialog/input-action-dialog.component.ts

```diff
@@ -84,4 +86,55 @@ export class InputActionDialog {
   close() {
     this.modal.close();
   }
+
+  checkIfJenkinsIsUpAndProceedURL() {
+    let url = 'https://jenkins.prod-preview.openshift.io';
+    let notification;
+    this.http.get(url)
+      .subscribe((response) => {
+        switch (response.status) {
+          case 200:
+            notification = {
+              message: `Got ${response.status}, Jenkins is up and running...`,
+              type: NotificationType.SUCCESS
+            };
+            this.invokeUrl(this.inputAction.proceedUrl);
+            break;
+          case 307:
+          case 302:
+            notification = {
+              message: `Got ${response.status}, Connecting to Jenkins.`,
+              type: NotificationType.INFO
+            };
+            break;
+          case 503:
+            notification = {
+              message: `Got ${response.status}, Cluster resources capacity is full. Waiting along.`,
+              type: NotificationType.INFO
+            };
+            break;
+          case 504:
+            notification = {
+              message: `Got ${response.status}, Request timed out. Trying again.`,
+              type: NotificationType.INFO
+            };
+            break;
+          case 202:
+            notification = {
+              message: `Got ${response.status}, Jenkins is currently idled. Please wait while we start it. waiting along...`,
+              type: NotificationType.WARNING
+            };
+            break;
+          default:
+            notification = {
+              message: `Got status ${response.status}`,
+              type: NotificationType.DANGER
+            };
+            break;
+        }
+        this.notifications.message(notification as Notification);
```

Comment:
Notice that now there is no return as it happens in if/else so infinite loop might occurs because break only occurs in switch context.

Reply:
This only sets the `notification` variable. There is nothing returned at all. The `default` sets it too. This looks correct to me.